otherlibs/dune-rpc-lwt/src/dune_rpc_lwt.ml

```diff
@@ -92,15 +92,20 @@ module V1 = struct
 
         let read_file s = Lwt_io.with_file ~mode:Input s Lwt_io.read
 
-        let readlink s = Lwt_unix.readlink s
+        let readlink s =
```

Comment:
I'm not sure how general this function is supposed to be, but there's a race here between `readlink` and somebody modifying the directory.
Probably not a problem, but if I had a choice between having the race in a shared library or at the call site I'd probably choose at the call site. (feel free to ignore if you disagree)

Reply:
How about an implementation that is race free but relies on exceptions:

```ocaml
try
  Some (Unix.readlink s)
with Unix.Unix_error (EINVAL, _, _) -> None
```

Reply:
I like that better, yeah. (I'm assuming here `Unix.readlink` is the blocking one, not lwt one?)

Reply:
Yes. For lwt, I will `Lwt_unix.readlink` and `Lwt.catch` respectively.

Reply:
Switched to this implementation